library/src/main/java/com/yausername/youtubedl_android/YoutubeDLOptions.java

```diff
@@ -9,40 +9,67 @@
 
 public class YoutubeDLOptions {
 
-    private Map<String, String> options = new LinkedHashMap<>();
+    private final Map<String, List<String>> options = new LinkedHashMap<>();
 
-    public YoutubeDLOptions addOption(@NonNull String key, @NonNull String value){
-        options.put(key, value);
+    public YoutubeDLOptions addOption(@NonNull String option, @NonNull String argument) {
+        if (!options.containsKey(option)) {
+            List<String> arguments = new ArrayList<>();
+            arguments.add(argument);
+            options.put(option, arguments);
+        } else {
+            options.get(option).add(argument);
+        }
         return this;
     }
 
-    public YoutubeDLOptions addOption(@NonNull String key, @NonNull Number value){
-        options.put(key, value.toString());
+    public YoutubeDLOptions addOption(@NonNull String option, @NonNull Number argument) {
+        if (!options.containsKey(option)) {
+            List<String> arguments = new ArrayList<>();
+            arguments.add(argument.toString());
+            options.put(option, arguments);
+        } else {
+            options.get(option).add(argument.toString());
+        }
         return this;
     }
 
-    public YoutubeDLOptions addOption(String key){
-        options.put(key, null);
+    public YoutubeDLOptions addOption(String option) {
+        if (!options.containsKey(option)) {
+            List<String> arguments = new ArrayList<>();
+            arguments.add("");
+            options.put(option, arguments);
+        } else {
+            options.get(option).add("");
+        }
         return this;
     }
 
-    public Object getOption(String key){
-        return options.get(key);
+    public String getArgument(String option) {
+        String argument = options.get(option).get(0);
```

Comment:
Here you ave to check if the option is contained in the map. Otherwise the first get will return null and the second will throw NullPointerException.